Contributor/runtime, first version (#105)
decentraland/documentation: content/contributor/runtime/components.md

`Transform` has a size of `44` bytes, with this exact layout:

```goat
.-----------------------.-------------------------------.-----------------------.-----------------.
|   x   |   y   |   z   |   x   |   y   |   z   |   w   |   x   |   y   |   z   | parent (uint32) |
'-----------------------'-------------------------------'-----------------------'-----------------'
╵  position (3x float)  ╵      rotation (4x float)      ╵   scale (3x float)    ╵          
```

This approach allows the runtime (especially in low-level code environments) to avoid intermediate serialization and deserialization steps. The memory where a `Transform` resides can be copied, shared and directly pointed to.

Coordinates can be fractionary, and thus use a 4-byte floating point number.

The `parent` field indicates that this Entity should be positioned relative to another one, identified by their ID. 

Since the null-value for `parent` is also the [`RootEntity`]({{< ref "/contributor/runtime/entities#RootEntity" >}}) ID (`0`), any `Transform` is relative to it by default.

---

> ###### `MeshRenderer` <small>[↗ source](https://github.com/decentraland/protocol/blob/ccb88d679f20c0e22840c324879d7b2535f6c9a6/proto/decentraland/sdk/components/mesh_renderer.proto#L7)</small> {#MeshRenderer}

Reviewer: We may want to make this links target=_blank

Author: Hmm, I would certainly open links in new tabs if I was browsing this page, but I think it's something we can leave to the user. Do you feel it makes an important difference?

Provides basic rendering behavior for an Entity.

It can be set to render a plane, a sphere, a cube or cylinder.

Entities with a [`GltfContainer`](#GltfContainer) are rendered according to the referenced model, and ignore this component.

---
###### `MeshCollider` <small>[↗ source](https://github.com/decentraland/protocol/blob/ccb88d679f20c0e22840c324879d7b2535f6c9a6/proto/decentraland/sdk/components/mesh_collider.proto#L15)</small> {#MeshCollider}

Provides basic collision behavior and mouse pointer detection for an Entity.

It can be set to behave like a plane, a sphere, a cube or cylinder.

> Entities with a [`GltfContainer`](#GltfContainer) detect collisions according to meshes defined in the 3D model, and ignore this component.

Reviewer: I think both components are completely compatible, it would be strange to see though.
```suggestion

```

---
###### `Material` <small>[↗ source](https://github.com/decentraland/protocol/blob/ccb88d679f20c0e22840c324879d7b2535f6c9a6/proto/decentraland/sdk/components/material.proto#L19)</small> {#Material}

Sets the texture, lighting, color and transparency properties of an Entity that also has the [`MeshRenderer`](#MeshRenderer) component.

It's a complex structure, but all fields have default values that can be left as they are, in order to only change a subset of the properties.

> Entities with a [`GltfContainer`](#GltfContainer) are rendered according to the referenced model, and ignore this component.

Reviewer: Same here

---
###### `GltfContainer` <small>[↗ source](https://github.com/decentraland/protocol/blob/ccb88d679f20c0e22840c324879d7b2535f6c9a6/proto/decentraland/sdk/components/gltf_container.proto#L7)</small> {#GltfContainer}

Attaches a 3D model to this Entity, given the [file path]({{< ref "/contributor/content/entities#files" >}}) of a `.gltf` asset in the scene's manifest.

> Since the model has its own meshes and materials, this component overrides any behavior from [`MeshRenderer`](#MeshRenderer), [`MeshCollider`](#MeshCollider) and [`Material`](#Material).

Reviewer: Same
```suggestion
Since the model has its own meshes and materials, this component overrides any behavior from [`MeshRenderer`](#MeshRenderer), [`MeshCollider`](#MeshCollider) and [`Material`](#Material).
```

---
###### `Billboard` <small>[↗ source](https://github.com/decentraland/protocol/blob/ccb88d679f20c0e22840c324879d7b2535f6c9a6/proto/decentraland/sdk/components/billboard.proto#L17)</small> {#Billboard}

> Makes an Entity automatically reorient its `Transform` to face the camera. As the name indicates, it's used to display in-game billboards and frequently combined with [`TextShape`](#TextShape).

Reviewer: ```suggestion
Makes an Entity automatically reorient its `Transform` ensuring that the entity is always facing towards the camera. As the name indicates, it's used to display in-game billboards and frequently combined with [`TextShape`](#TextShape).

Billboards can be applied to parent entities, and the children will be re-oriented respecting their local `Transform`. In other words, child entities can be re-oriented inside a Billboard e.g. to rotate it another extra 180 degrees.
```

Author: Why this? Is it an alternative way to phrase it, or is this implying a technical detail I'm not aware of?

Reviewer: In a second read, it is the same
